model-providers/openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/runtime/AzureOpenAiRecorder.java

```diff
@@ -295,6 +298,10 @@ private LangChain4jAzureOpenAiConfig.AzureAiConfig correspondingAzureOpenAiConfi
 
     private void throwIfApiKeysNotConfigured(String apiKey, String adToken, String configName) {
         if ((apiKey != null) == (adToken != null)) {
+            ArcContainer container = Arc.container();
+            if (container != null && container.instance(ModelAuthProvider.class).isAvailable()) {
+                return;
+            }
```

Comment:
Hi @geoand I only would like to avoid the start up failure when no key or token is pre-configured... 
I guess using `BeanContainer` is the way to go

Reply:
> That leaves only the first option

This should be entirely possible because there are build items that allow you to see what kinds of beans are available

Reply:
But there is also another option. You can make this bean have an optional injection point to `ModelAuthProvider` and check if the bean exists in the recorder. We do that in various places

Reply:
@geoand Which bean would have the injection point, can you link me to some example please ?

Yeah I see examples how to do it at build time but what you suggested in the last comment appears to be much simpler to implement :-)

Reply:
I see, `addInjectionPoint`...

Reply:
@geoand I think I made it work. The quality of support around the synthetic bean creation is just insanely good...
It may need a bit more tweaking, to check that either generic and ModelName qualified injection points are resolvable, but it just works, thanks for the hint

Reply:
@geoand I've added 2 injection points, that should all be covered now, please check next week what needs to be tuned further, cheers !

Reply:
I don't think you need two injection points. The way I understand it, you only need one with the corresponding bean (with or without the qualifier)

Reply:
@geoand Sure, I'll test it to check what happens with a single point if `ModelProvider` is annotated with say `@ModelName("ollama")`... I'll let you know

Reply:
@geoand You are right, only a single injection point is enough to check if `ModelAuthProvider` is available, it detects all types of providers. There is a possibility with this check that if only a provider say for `ollama` is available then the failure will be delayed up until the `ModelAuthProvider.resolve` time, which is probably not going to happen often...